nilearn/plotting/html_stat_map.py

```diff
@@ -116,6 +135,7 @@ def view_stat_map(stat_map_img, threshold=None, bg_img='MNI152',
         Jupyter notebook.
 
     """
+    _view_stat_map_change_warning()
```

Comment:
There is no use to create a function inside a function to do this. Having the warning here would be better, as it would simplify the stack, and make the warning easier to understand.

Reply:
The function has been created outside, not within this function.

This is better, cause the function's name tells exactly what is being done here. I also don't want to add 4 lines to a function that's already 10-15 lines long; way more than Python standard recommend.

Having a separate function is generally cleaner in stack traces because then I don't have to examine every line to trace what is going wrong. In many CIs the function names are given in the trace but the line number is not and writing many small function obviates that problem.

I aim to refactor the existing functions and I want the new contributors to see that more and more. I don't want them to feel it is okay to insert "just a teeny more simple functionality in an existing function". A few insertions and then we have 20 (or more) line long functions.